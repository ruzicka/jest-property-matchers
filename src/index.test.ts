import * as Chance from 'chance'
import generateMatcher from './index'

const chance = new Chance()

describe('matchers', () => {

  describe('$number', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        id: chance.integer(),
      }

      const matcher = generateMatcher({
        $number: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        id: chance.integer(),
      }

      const matcher = generateMatcher({
        $number: 'id',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        id: chance.integer(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $number: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        id: chance.integer(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $number: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        id: chance.integer(),
        _id: chance.integer(),
      }

      const matcher = generateMatcher({
        $number: ['id', '_id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        id: chance.integer(),
        _id: chance.integer(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $number: ['id', '_id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('$numericString', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        id: chance.integer().toString(),
      }

      const matcher = generateMatcher({
        $numericString: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        id: chance.integer().toString(),
      }

      const matcher = generateMatcher({
        $numericString: 'id',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        id: chance.integer().toString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $numericString: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        id: chance.integer().toString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $numericString: ['id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        id: chance.integer().toString(),
        _id: chance.integer().toString(),
      }

      const matcher = generateMatcher({
        $numericString: ['id', '_id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        id: chance.integer().toString(),
        _id: chance.integer().toString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $numericString: ['id', '_id'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('$dateTime', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        createdAt: chance.date(),
      }

      const matcher = generateMatcher({
        $dateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        createdAt: chance.date(),
      }

      const matcher = generateMatcher({
        $dateTime: 'createdAt',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $dateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $dateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        createdAt: chance.date(),
        created_at: chance.date(),
      }

      const matcher = generateMatcher({
        $dateTime: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        created_at: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $dateTime: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('$date', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        createdAt: chance.date(),
      }

      const matcher = generateMatcher({
        $date: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        createdAt: chance.date(),
      }

      const matcher = generateMatcher({
        $date: 'createdAt',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $date: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $date: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        createdAt: chance.date(),
        created_at: chance.date(),
      }

      const matcher = generateMatcher({
        $date: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        createdAt: chance.date(),
        created_at: chance.date(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $date: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('$stringDateTime', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
      }

      const matcher = generateMatcher({
        $stringDateTime: 'createdAt',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
        created_at: chance.date().toISOString(),
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString(),
        created_at: chance.date().toISOString(),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('$stringDate', () => {

    test('single random property (defined in array) only', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
      }

      const matcher = generateMatcher({
        $stringDate: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property only', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
      }

      const matcher = generateMatcher({
        $stringDate: 'createdAt',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property (defined in array) and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDate: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('single random property and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDate: ['createdAt'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties only', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
        created_at: chance.date().toISOString().substr(0, 10),
      }

      const matcher = generateMatcher({
        $stringDate: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('multiple random properties and some other constant props', () => {
      const obj = {
        createdAt: chance.date().toISOString().substr(0, 10),
        created_at: chance.date().toISOString().substr(0, 10),
        num: 10,
        str: 'test string',
      }

      const matcher = generateMatcher({
        $stringDate: ['createdAt', 'created_at'],
      })

      expect(obj).toMatchSnapshot(matcher)
    })
  })

  describe('more complex scenarios', () => {

    test('combination of all supported props', () => {
      const obj = {
        createdAt: chance.date(),
        created_at: chance.date(),
        createdAtStr: chance.date().toISOString(),
        created_at_str: chance.date().toISOString(),
        bornAt: chance.date(),
        bornAtStr: chance.date().toISOString().substr(0, 10),
        id: chance.integer(),
        _id: chance.integer(),
        bigIntId: chance.integer({ min: 0 }).toString(),
        otherProp: [10, 20],
        someOtherProp: 'str value',
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAtStr', 'created_at_str'],
        $stringDate: 'bornAtStr',
        $number: ['id', '_id'],
        $numericString: 'bigIntId',
        $dateTime: ['createdAt', 'created_at'],
        $date: 'bornAt',
      })

      expect(obj).toMatchSnapshot(matcher)
    })

    test('nested objects', () => {
      const obj = {
        createdAtStr: chance.date().toISOString(),
        created_at_str: chance.date().toISOString(),
        bornAtStr: chance.date().toISOString().substr(0, 10),
        id: chance.integer(),
        _id: chance.integer(),
        otherProp: [10, 20],
        subObject: {
          bigIntId: chance.integer({ min: 0 }).toString(),
          createdAt: chance.date(),
          created_at: chance.date(),
          bornAt: chance.date(),
          someOtherProp: 'str value',
        },
      }

      const matcher = generateMatcher({
        $stringDateTime: ['createdAtStr', 'created_at_str'],
        $stringDate: 'bornAtStr',
        $number: ['id', '_id'],
        subObject: {
          $numericString: 'bigIntId',
          $dateTime: ['createdAt', 'created_at'],
          $date: 'bornAt',
        },
      })

      expect(obj).toMatchSnapshot(matcher)
    })

  })

})
