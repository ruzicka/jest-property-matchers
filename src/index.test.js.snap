// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchers $date multiple random properties and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "created_at": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $date multiple random properties only 1`] = `
Object {
  "createdAt": Any<Date>,
  "created_at": Any<Date>,
}
`;

exports[`matchers $date single random property (defined in array) and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $date single random property (defined in array) only 1`] = `
Object {
  "createdAt": Any<Date>,
}
`;

exports[`matchers $date single random property and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $date single random property only 1`] = `
Object {
  "createdAt": Any<Date>,
}
`;

exports[`matchers $dateTime multiple random properties and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "created_at": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $dateTime multiple random properties only 1`] = `
Object {
  "createdAt": Any<Date>,
  "created_at": Any<Date>,
}
`;

exports[`matchers $dateTime single random property (defined in array) and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $dateTime single random property (defined in array) only 1`] = `
Object {
  "createdAt": Any<Date>,
}
`;

exports[`matchers $dateTime single random property and some other constant props 1`] = `
Object {
  "createdAt": Any<Date>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $dateTime single random property only 1`] = `
Object {
  "createdAt": Any<Date>,
}
`;

exports[`matchers $number multiple random properties and some other constant props 1`] = `
Object {
  "_id": Any<Number>,
  "id": Any<Number>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $number multiple random properties only 1`] = `
Object {
  "_id": Any<Number>,
  "id": Any<Number>,
}
`;

exports[`matchers $number single random property (defined in array) and some other constant props 1`] = `
Object {
  "id": Any<Number>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $number single random property (defined in array) only 1`] = `
Object {
  "id": Any<Number>,
}
`;

exports[`matchers $number single random property and some other constant props 1`] = `
Object {
  "id": Any<Number>,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $number single random property only 1`] = `
Object {
  "id": Any<Number>,
}
`;

exports[`matchers $numericString multiple random properties and some other constant props 1`] = `
Object {
  "_id": StringMatching /\\[0-9\\]\\+/u,
  "id": StringMatching /\\[0-9\\]\\+/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $numericString multiple random properties only 1`] = `
Object {
  "_id": StringMatching /\\[0-9\\]\\+/u,
  "id": StringMatching /\\[0-9\\]\\+/u,
}
`;

exports[`matchers $numericString single random property (defined in array) and some other constant props 1`] = `
Object {
  "id": StringMatching /\\[0-9\\]\\+/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $numericString single random property (defined in array) only 1`] = `
Object {
  "id": StringMatching /\\[0-9\\]\\+/u,
}
`;

exports[`matchers $numericString single random property and some other constant props 1`] = `
Object {
  "id": StringMatching /\\[0-9\\]\\+/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $numericString single random property only 1`] = `
Object {
  "id": StringMatching /\\[0-9\\]\\+/u,
}
`;

exports[`matchers $string multiple random properties and some other constant props 1`] = `
Object {
  "num": 10,
  "str": "test string",
  "stringProp": Any<String>,
  "string_prop": Any<String>,
}
`;

exports[`matchers $string multiple random properties only 1`] = `
Object {
  "stringProp": Any<String>,
  "string_prop": Any<String>,
}
`;

exports[`matchers $string single random property (defined in array) and some other constant props 1`] = `
Object {
  "num": 10,
  "str": "test string",
  "stringProp": Any<String>,
}
`;

exports[`matchers $string single random property (defined in array) only 1`] = `
Object {
  "stringProp": Any<String>,
}
`;

exports[`matchers $string single random property and some other constant props 1`] = `
Object {
  "num": 10,
  "str": "test string",
  "stringProp": Any<String>,
}
`;

exports[`matchers $string single random property only 1`] = `
Object {
  "stringProp": Any<String>,
}
`;

exports[`matchers $stringDate multiple random properties and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "created_at": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDate multiple random properties only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "created_at": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
}
`;

exports[`matchers $stringDate single random property (defined in array) and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDate single random property (defined in array) only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
}
`;

exports[`matchers $stringDate single random property and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDate single random property only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
}
`;

exports[`matchers $stringDateTime multiple random properties and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "created_at": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDateTime multiple random properties only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "created_at": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
}
`;

exports[`matchers $stringDateTime single random property (defined in array) and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDateTime single random property (defined in array) only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
}
`;

exports[`matchers $stringDateTime single random property and some other constant props 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "num": 10,
  "str": "test string",
}
`;

exports[`matchers $stringDateTime single random property only 1`] = `
Object {
  "createdAt": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
}
`;

exports[`matchers more complex scenarios combination of all supported props 1`] = `
Object {
  "_id": Any<Number>,
  "bigIntId": StringMatching /\\[0-9\\]\\+/u,
  "bornAt": Any<Date>,
  "bornAtStr": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "createdAt": Any<Date>,
  "createdAtStr": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "created_at": Any<Date>,
  "created_at_str": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "id": Any<Number>,
  "otherProp": Array [
    10,
    20,
  ],
  "someOtherProp": "str value",
}
`;

exports[`matchers more complex scenarios nested objects 1`] = `
Object {
  "_id": Any<Number>,
  "bornAtStr": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\d\\$/u,
  "createdAtStr": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "created_at_str": StringMatching /\\^\\\\d\\{4\\}-\\[01\\]\\\\d-\\[0-3\\]\\\\dT\\[0-2\\]\\\\d:\\[0-5\\]\\\\d:\\[0-5\\]\\\\d\\\\\\.\\\\d\\+\\(\\[\\+-\\]\\[0-2\\]\\\\d:\\[0-5\\]\\\\d\\|Z\\)\\$/u,
  "id": Any<Number>,
  "otherProp": Array [
    10,
    20,
  ],
  "subObject": Object {
    "bigIntId": StringMatching /\\[0-9\\]\\+/u,
    "bornAt": Any<Date>,
    "createdAt": Any<Date>,
    "created_at": Any<Date>,
    "someOtherProp": "str value",
  },
}
`;
